apollo-federation/src/sources/connect/validation/http/url.rs

```diff
@@ -2,62 +2,126 @@ use std::fmt::Display;
 use std::ops::Range;
 use std::str::FromStr;
 
+use apollo_compiler::ast::FieldDefinition;
+use apollo_compiler::ast::Type;
 use apollo_compiler::ast::Value;
 use apollo_compiler::parser::LineColumn;
 use apollo_compiler::parser::SourceMap;
+use apollo_compiler::schema::Component;
+use apollo_compiler::schema::ExtendedType;
 use apollo_compiler::Node;
+use apollo_compiler::Schema;
 use url::Url;
 
 use crate::sources::connect::url_template;
+use crate::sources::connect::url_template::VariableType;
 use crate::sources::connect::validation::coordinates::HttpMethodCoordinate;
 use crate::sources::connect::validation::require_value_is_str;
 use crate::sources::connect::validation::Code;
 use crate::sources::connect::validation::Message;
 use crate::sources::connect::URLTemplate;
+use crate::sources::connect::Variable;
 
 pub(crate) fn validate_template(
-    value: &Node<Value>,
     coordinate: HttpMethodCoordinate,
+    schema: &Schema,
+) -> Result<URLTemplate, Vec<Message>> {
+    let (template, str_value) = match parse_template(coordinate, &schema.sources) {
+        Ok(tuple) => tuple,
+        Err(message) => return Err(vec![message]),
+    };
+    let mut messages = Vec::new();
+    if let Some(base) = template.base.as_ref() {
+        messages.extend(
+            validate_base_url(
+                base,
+                coordinate,
+                coordinate.node,
+                str_value,
+                &schema.sources,
+            )
+            .err(),
+        );
+    }
+
+    for variable in template.path_variables() {
+        match validate_variable(variable, str_value, coordinate, schema) {
+            Err(err) => messages.push(err),
+            Ok(Some(ty)) if !ty.is_non_null() => {
+                messages.push(Message {
+                    code: Code::NullablePathVariable,
+                    message: format!(
+                        "Variables in path parameters should be non-null, but {coordinate} contains `{{{variable}}}` which is nullable.\
```

Comment:
```suggestion
                        "Variables in path parameters should be non-null, but {coordinate} contains `{{{variable}}}` which is nullable. \
```
adds a space between sentences